### DISCUSSION TITLE
GUI for Creating and Editing Pipelines

### DISCUSSION THREAD

**Discussion by FabianPost** [kedro-org/kedro-viz#1829] [Ideas]
## Description
Developing Pipelines in Node editors enables faster pipeline creation and better intuitive programming, especially for data-scientists without software engineering background. Similiar to LabView and MATLAB-Simulink, I seek for a GUI to create and modify pipelines that compiles the changes directly into kedro python code and saves it inside the project.

## Context
Kedro viz allows data scientists to inspect and overview pipelines. However, it does not enable creating oder editing those. Editing pipelines should be easier when you do not have to track input and output names of complex pipelines, but instead simply connect outputs with inputs and name the connection (i.e., input and output) on demand only.
This enables new users to immediatly get started with kedro and allows them to dive into the code base of their own pipelines when neccesary.

## Possible Implementation
The editing functions could probably be added to kedro viz.

## Possible Alternatives
A seperated Node Editor, e.g., based on Dear GUI, could be used as a compiler.

**Comment by merelcht** [comment]
Hi @FabianPost, thanks for creating this issue! We've heard this request before and will explore it in more depth. We'll assess user needs and feasibility.

**Comment by yetudada** [comment]
From #2055:

- Make Kedro more interactive then just CLI maybe somehow incorporate with Viz.
- New CLI/UI for interactivity with pipeline for "low tech" users
- Full control over Pipeline build (code authoring) in Viz

**Comment by hienchu** [comment]
Is there any ongoing work on the feature?

**Comment by astrojuanlu** [comment]
I want to note that GUIs like these rely heavily on reusable components. Otherwise it's just about connecting boxes, in which case we could think of tackling https://github.com/kedro-org/kedro/issues/2577 instead

**Comment by astrojuanlu** [comment]
@rashidakanchwala early user feedback on the "pipeline editor" feature
